cocotb/handle.py

```diff
@@ -104,12 +104,13 @@ def __len__(self):
         return self._len
 
     def __eq__(self, other):
+        """Equality comparator for handles
 
-        # Permits comparison of handles i.e. if clk == dut.clk
-        if isinstance(other, SimHandleBase):
-            if self._handle == other._handle:
-                return 0
-            return 1
+        IE if clk == dut.clk
+        """
+        if not isinstance(other, SimHandleBase):
+            return NotImplemented
+        return self._handle == other._handle
 
     def __ne__(self, other):
         return not self.__eq__(other)
```

Comment:
This will do the wrong thing if `__eq__` returns `NotImplemented`

Reply:
What's that?

Reply:
You need to have __ne__ return Not Implemented if __eq__ does.

The test I proposed above should catch this.